src/formElementPlugins/fileUpload/src/ApplicationView.tsx

```diff
@@ -50,6 +53,7 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
     fileExtensions,
     fileSizeLimit,
     default: defaultValue,
+    showDocumentModal = preferences.showDocumentModal,
```

Comment:
The system-wide preference for "showDocumentModal" can be overridden for each FileUpload element if desired.